src/Foundation/Config.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation;
+
+use ArrayAccess;
+use Illuminate\Support\Arr;
+use InvalidArgumentException;
+
+class Config implements ArrayAccess
+{
+    private $data;
+
+    public function __construct(array $data)
+    {
+        $this->data = $data;
+
+        $this->requireKeys('url');
+    }
+
+    public function url(): string
+    {
+        return $this->data['url'];
+    }
+
+    public function inDebugMode(): bool
+    {
+        return $this->data['debug'] ?? false;
+    }
+
+    public function inMaintenanceMode(): bool
+    {
+        return $this->data['offline'] ?? false;
+    }
+
+    private function requireKeys(...$keys)
+    {
+        foreach ($keys as $key) {
+            if (! array_key_exists($key, $this->data)) {
+                throw new InvalidArgumentException(
+                    "Configuration is invalid without a $key key"
+                );
+            }
+        }
+    }
+
+    public function offsetGet($offset)
+    {
+        return Arr::get($this->data, $offset);
+    }
+
+    public function offsetExists($offset)
+    {
+        return Arr::has($this->data, $offset);
+    }
+
+    public function offsetSet($offset, $value)
+    {
+        // No mutation allowed
```

Comment:
Because it's not a real mutation anyway - objects of this class represent the values written in the `config.php` file. Do you have another example? Or can you point me to the code for your redis example?

Reply:
🤔 I think I've mistaken the flarum config with the laravel one. Ignore that statement 😂 

I'm okay with your proposal.

Reply:
Done.